### ISSUE TITLE
SIGSEGV on Ubuntu 16.04, pthread

### ISSUE THREAD

**Issue by FM33** [Beep6581/RawTherapee#3814]
Hello,
Since recent versions, I encounter frequent crash after 5-10 min of use. It never occurs doing the same thing, just moving sliders, changing values or curves.

```
Version: 5.0-r1-gtk3-400-g92a46e9
Branch: dev
Commit: 92a46e9
Commit date: 2017-04-09
Compiler: cc 5.3.1
Processor: x86_64
System: Linux (Ubuntu 16.04)
Bit depth: 64 bits
Gtkmm: V3.18.0
Build type: Debug
Build flags: -std=c++11 -Wno-deprecated-declarations -Wno-unused-result -std=c++11 -march=native -Werror=unused-label -fopenmp -Werror=unknown-pragmas -Wall -Wno-unused-result -Wno-deprecated-declarations -g
Link flags:  -march=native
OpenMP support: ON
MMAP support: ON
```
```
Thread 867 "rawtherapee" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff7b188700 (LWP 25221)]
0x00007ffff6fccd65 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
(gdb) bt full
#0  0x00007ffff6fccd65 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#1  0x00007ffff6fcd099 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#2  0x00007ffff2d8c439 in __nptl_deallocate_tsd () at pthread_create.c:175
        data = <optimized out>
        level2 = 0x7fff7b188a10
        idx = <optimized out>
        cnt = 0
#3  0x00007ffff2d8d870 in __nptl_deallocate_tsd () at pthread_create.c:326
No locals.
#4  start_thread (arg=0x7fff7b188700) at pthread_create.c:346
        pd = 0x7fff7b188700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140735258593024, 
                -4415184257679638046, 0, 140737488343599, 8388608, 202875872, 
                4414912467987841506, 4415208782798373346}, 
              mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, 
            data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
        pagesize_m1 = <optimized out>
        sp = <optimized out>
        freesize = <optimized out>
---Type <return> to continue, or q <return> to quit---
        __PRETTY_FUNCTION__ = "start_thread"
#5  0x00007ffff2ac382d in clone ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
No locals.
```

**Comment by ff2000** [comment]
Could you check https://github.com/Beep6581/RawTherapee/commit/028edd5bd9d08f5d0dde8af409db56b7a17f1707? It says "startup crashes" but I also saw crashes recently while editing. Not tested for too much but it is very stable - not a single crash so far.

**Comment by Beep6581** [comment]
Actually please pull the latest dev and test there again (commit 56e4114c or newer).

**Comment by FM33** [comment]
2 hrs without a crash. It seems to be solved for me with 56e4114.
